cloud/envs/gcp.py

```diff
@@ -192,12 +192,13 @@ def add(self, *args, **kwargs):
         return tpu
     return super().add(*args, **kwargs)
 
-  def get(self, preemptible=True):
+  def get(self, preemptible=True, with_name=None):
     for tpu in self.resources:
-      if tpu.usable:
+      if tpu.usable and not with_name or tpu.name == with_name:
```

Comment:
this statement is super ambiguous. `and` has precedence over `or` so this is really saying `if (tpu.usable and not with_name) or tpu.name == with_name` which isn't what you intended I believe

Reply:
also let's change `with_name` to just `name`